### ISSUE TITLE
Temp DLL files prefixed with ~ when editor is open

### ISSUE THREAD

**Issue by levidavidmurray** [godot-jolt/godot-jolt#690]
**Window 11**
**Godot v4.2.rc1.official**
**godot-jolt 0.10.0**

I didn't realize that I had initially committed this temp file, and was confused as to why git was reporting a deleted file after closing the editor. I can add it to my `.gitignore` easily enough, but I'm just wondering if this is expected (or known about) as I haven't been able to find any mention of it.

I first noticed it with this `~.dll~[...].TMP` file that was somehow created. I haven't been able to reproduce the `.TMP` file, but it's worth noting that it persisted even after closing the editor (regular `~.dll` was deleted).

![image](https://github.com/godot-jolt/godot-jolt/assets/46084870/18fc7d02-2abb-4934-bd05-0a01926a2eb0)

This is after deleting the `~.dll` file from remote:

### Editor Closed
![image](https://github.com/godot-jolt/godot-jolt/assets/46084870/69016ebb-3b40-478f-8efe-3fc7a4eb1a8a)

### Editor Open
![image](https://github.com/godot-jolt/godot-jolt/assets/46084870/2323b7f7-4601-4421-93be-01f4d8b2057d)

I suppose this could very well be something specific to GDExtensions in Godot 4.2, but I figured I'd start here to see if you're aware of this.

✌️

**Comment by mihe** [comment]
> I suppose it's likely this is something specific to GDExtensions in Godot 4.2.rc1

Indeed. This is to facilitate hot-reloading of extension DLLs on Windows, as seen in godotengine/godot#80188 and godotengine/godot#80284, where Godot needs to first copy any extension DLLs to a new file and then load that file instead, due to Windows locking any files that are currently open/loaded, which prevents them from being overwritten.

I'm not entirely sure where the `*.TMP` file comes from, but I would wager that it likely comes from Godot's "Safe Save on Backup then Rename" editor setting, which you'll find under "Editor Settings / FileSystem / On Save", which causes Godot to always write (or in this copy) to a temporary file first before moving that to its final destination.

> and it persisted even after closing the editor

You should ideally report this to [godotengine/godot](https://github.com/godotengine/godot), as failure to remove these temporary files is a bug.
